helix-core/src/search.rs

```diff
@@ -61,3 +61,184 @@ pub fn find_nth_prev(
 
     Some(pos)
 }
+
+use crate::movement::Direction;
+use regex_automata::{dense, DenseDFA, Error as RegexError, DFA};
+use std::ops::Range;
+
+#[derive(Debug)]
+pub struct Searcher {
+    /// Locate end of match searching right.
+    right_fdfa: DenseDFA<Vec<usize>, usize>,
+    /// Locate start of match searching right.
+    right_rdfa: DenseDFA<Vec<usize>, usize>,
+
+    /// Locate start of match searching left.
+    left_fdfa: DenseDFA<Vec<usize>, usize>,
+    /// Locate end of match searching left.
+    left_rdfa: DenseDFA<Vec<usize>, usize>,
+}
+
+impl Searcher {
+    pub fn new(pattern: &str) -> Result<Searcher, RegexError> {
+        // Check case info for smart case
+        let has_uppercase = pattern.chars().any(|c| c.is_uppercase());
```

Comment:
N.B. This is how ripgrep does smart case detection: https://github.com/BurntSushi/ripgrep/blob/9f924ee187d4c62aa6ebe4903d0cfc6507a5adb5/crates/regex/src/ast.rs

Reply:
I spotted that, should probably copy that behavior over. The current detection matches alacritty, might be worth pointing out there as well.